src/Lime.Transport.WebSocket/PipeWebSocketTransport.cs

```diff
@@ -0,0 +1,251 @@
+            }
+            catch (InvalidOperationException)
+            {
+                await CloseWithTimeoutAsync().ConfigureAwait(false);
+                throw;
+            }
+        }
+
+        protected override Task PerformOpenAsync(Uri uri, CancellationToken cancellationToken)
+        {
+            return _envelopePipe.StartAsync(cancellationToken);
+        }
+
+        /// <summary>
+        /// Closes the transport, implementing the websocket close handshake.
+        /// </summary>
+        /// <param name="cancellationToken"></param>
+        /// <returns></returns>
+        /// <exception cref="InvalidOperationException">The listener is not active</exception>
+        protected override async Task PerformCloseAsync(CancellationToken cancellationToken)
+        {
+            await _closeSemaphore.WaitAsync(cancellationToken);
+            try
+            {
+                await _envelopePipe.StopAsync(cancellationToken);
+                
+                // Awaits for the client to send the close connection frame.
+                // If the session was clearly closed, the client should received the finished envelope and is closing the connection.
+                using (var cts = new CancellationTokenSource(CloseTimeout))
+                using (var linkedCts = CancellationTokenSource.CreateLinkedTokenSource(cancellationToken, cts.Token))
+                {
+                    if (WebSocket.State == WebSocketState.Open ||
+                        WebSocket.State == WebSocketState.CloseReceived)
+                    {
+                        if (_closeGracefully)
+                        {
+                            await
+                                WebSocket.CloseAsync(CloseStatus, CloseStatusDescription, linkedCts.Token)
+                                    .ConfigureAwait(false);
+                        }
+                        else
+                        {
+                            await
+                                WebSocket.CloseOutputAsync(CloseStatus, CloseStatusDescription, linkedCts.Token)
+                                    .ConfigureAwait(false);
+                        }
+                    }
+                }
+            }
+            finally
+            {
+                _closeSemaphore.Release();
+            }
+        }
+
+        public override bool IsConnected => WebSocket.State == WebSocketState.Open ||
+                                            WebSocket.State == WebSocketState.CloseReceived; // We need to consider the CloseReceived status here to make the channel call the CloseAsync method.
+
+        public override string LocalEndPoint
+        {
+            get
+            {
+                try
+                {
+                    try
+                    {
+                        return WebSocket.AsDynamic()._innerStream?._context?.Request?.LocalEndPoint?.ToString();
```

Comment:
Now that this project targets netstandard2.1, this compatilibity code can be removed, can't it?

Reply:
nope...